data_reader.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"os"
+)
+
+type DataContent struct {
+	StartHeight   int64    `json:"start_height"`
+	HeaderStrings []string `json:"header_strings"`
+}
+
+func ReadJson(jsonFilePath string) (int64, []string, error) {
+	jsonFile, err := os.Open(jsonFilePath)
+	if err != nil {
+		log.Fatalf("Failed to open file: %v", err)
+		return 0, nil, err
+	}
+	defer jsonFile.Close()
+
+	// Read the file contents
+	byteValue, err := io.ReadAll(jsonFile)
+	if err != nil {
+		log.Fatalf("Failed to read file: %v", err)
```

Comment:
Don't use fatal logs if you return errors. It will fail the program immediately. Log in main, not in this function.